quartz/plugins/transformers/frontmatter.ts

```diff
@@ -73,6 +73,18 @@ export const FrontMatter: QuartzTransformerPlugin<Partial<Options>> = (userOpts)
 
             const socialImage = coalesceAliases(data, ["socialImage", "image", "cover"])
 
+            const created = coalesceAliases(data, ["date", "created"])
+            if (created) data.created = created
+            const modified = coalesceAliases(data, [
+              "lastmod",
+              "updated",
+              "last-modified",
+              "modified",
+            ])
+            if (modified) data.modified = modified
+            const published = coalesceAliases(data, ["publishDate", "published", "date"])
```

Comment:
unlike all the other `coalesceAliases` in this file, the aliases for `created`, `modified`, and `published` have higher priority than the original property. i.e. the existing value in the `published` property will be replaced by one in the `publishDate` property. is this intentional?

Reply:
`coalesceAliases` will return the first non-undefined, non-null item.

for example, if the value `publishDate` exists, then it will do early `return`.

This is essentially similar to `||=`

Reply:
Correct me if I'm wrong, but don't we have `=` semantics rather than `||=`:

```typescript
// code
const published = coalesceAliases(data, ["publishDate", "published", "date"])
if (published) data.published = published

// expected behavior equivalent to
data.published ||= coalesce(data.publishDate, data.date)

// actual behavior equivalent to
data.published = coalesce(data.publishDate, data.published, data.date)
```

i.e., assuming:

```yaml
---
...
publishDate: 1980-10-25
published: 2020-01-01
...
---
```

wouldn't we get:

```typescript
frontmatter.data.published == '1980-10-25'
```

Reply:
> i.e., assuming:
> 
> ```yaml
> ---
> ...
> publishDate: 1980-10-25
> published: 2020-01-01
> ...
> ---
> ```
> 
> wouldn't we get:
> 
> ```ts
> frontmatter.data.published == '1980-10-25'
> ```

So your suggestion is to swap `publishDate` and `published` in the published frontmatter values array?

Reply:
seems like it

Reply:
I updated to align with previous behaviour. Thanks for the review